dietpi/dietpi-software

```diff
@@ -17651,6 +17653,57 @@ List of installed software and their online documentation URLs:
 			# Load all automation vars
 			FirstRun_Automation_Init
 
+			if (( $AUTOINSTALL_RESTORE )); then
+
+					# Find unmounted drives
+					lsblk --noheadings --raw -o NAME,MOUNTPOINT | mawk '$1~/s.*[0-9]/ && $2 == ""' > .lsblk_out_tmp
+					i=0
+					while read -r line
+					do
+
+						# Mount drives to temporary mount points
+						mkdir "/mnt/dietpi-backup$i"
+						mount "/dev/$line" "/mnt/dietpi-backup$i"
+
+						i++
+
+					done < .lsblk_out_tmp
+					rm .lsblk_out_tmp
+
+					mapfile -t alist < <(find /mnt -type f -name "$BACKUP_STATS_FILENAME")
+
+					if [[ $AUTOINSTALL_RESTORE == 1 ]]; then
+						# Do we have any results?
+						if (( ${alist[0]} )); then
+							# Create List for Whiptail
+							G_WHIP_MENU_ARRAY=()
+							for i in "${alist[@]}"
+							do
+
+								last_backup_date=$(sed -n '/ompleted/s/^.*: //p' "$i" | tail -1) # Date of last backup for this backup
+								backup_directory=${i%/$BACKUP_STATS_FILENAME} # Backup directory (minus the backup file), that we can use for target backup directory.
+								G_WHIP_MENU_ARRAY+=("$backup_directory" ": $last_backup_date")
+
+							done
+
+							G_WHIP_MENU 'Please select a previous backup to use:'
+							/boot/dietpi/dietpi-backup -1 "$G_WHIP_RETURNED_VALUE"
+
+						else
+
+							G_WHIP_MSG 'No previous backups were found in /mnt/*'
+							return
+						fi
+					elif [[ $AUTOINSTALL_RESTORE == 2 ]]; then
+						# Do we have any results?
+						if (( ${alist[0]} )); then
+							# Restore first found backup
+							/boot/dietpi/dietpi-backup -1 /mnt/dietpi-backup0
+						fi
```

Comment:
Probably we should print an error message when no backup as been found?
```
else
	G_DIETPI-NOTIFY 1 'DietPi-Backup auto-restore was selected but no backup has been found in /mnt/*.'
fi
```

And another idea: Shall we unset the `dietpi.txt` option so that, if for any reason first run code is executed again, it won't again non-interactively restore a backup? At least it could be degraded from `=2` to `=1`, just to be failsafe with such a step?

Reply:
at least to inform the user that no backup was found and therefore the non-interactively restore will be unset.

Reply:
In case of unsetting the setting, I meant to do this in every case, also after a successful backup, just to rule out that it can be executed a second time on that system accidentally. We could also wrap that command into an info:
```
G_EXEC_DESC='Disabling DietPi-Backup auto-restore to assure it cannot accidentally run a second time' G_EXEC sed -i '/^[[:blank:]]*AUTO_SETUP_BACKUP_RESTORE=/cAUTO_SETUP_BACKUP_RESTORE=0/' /boot/dietpi.txt
```

Reply:
understood. Good idea